### PR TITLE
Fix missing artists on Apple Music playlist tracks

### PR DESCRIPTION
**Pull request by MarvinSchenkel**
# What does this implement/fix?

Tracks of an Apple Music library playlist lose their artists when those artists are not in the user's library: Apple returns the `artists` relationship with empty `data`, the parser took that branch and never fell back to the `artistName` attribute. Downstream this broke the Music Quiz (guess-the-song rounds intermittently failed with "Not enough distractors to build suggestions" because the artistless source pool was discarded), but it affects any consumer of these tracks.

- Fall back to `artistName` when the artists relationship carries no data
- Bump `PARSED_ITEM_CACHE_CHECKSUM` so cached artistless tracks are re-parsed
- Add a regression test for the empty-relationship shape

**Related issue (if applicable):**

- related issue N/A

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue) — `bugfix`
- [ ] New feature (non-breaking change which adds functionality) — `new-feature`
- [ ] Enhancement to an existing feature — `enhancement`
- [ ] New music/player/metadata/plugin provider — `new-provider`
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected) — `breaking-change`
- [ ] Refactor (no behaviour change) — `refactor`
- [ ] Documentation only — `documentation`
- [ ] Maintenance / chore — `maintenance`
- [ ] CI / workflow change — `ci`
- [ ] Dependencies bump — `dependencies`

## Checklist

- [x] The code change is tested and works locally.
- [x] `pre-commit run --all-files` passes.
- [x] `pytest` passes, and tests have been added/updated under `tests/` where applicable.
- [ ] For changes to shared models, the companion PR in `music-assistant/models` is linked.
- [ ] For changes affecting the UI, the companion PR in `music-assistant/frontend` is linked.
- [x] I have read and complied with the project's [AI Policy](https://github.com/music-assistant/.github/blob/main/AI_POLICY.md) for any AI-assisted contributions.
- [ ] I have [raised a PR against the documentation repository](https://github.com/music-assistant/music-assistant.io/blob/main/CONTRIBUTING.md) targeting the main or beta branch as appropriate.

### CODE REVIEW

File: music_assistant/providers/apple_music/constants.py [hunk 1]
```diff
@@ -44,4 +44,4 @@
 # bust cached api responses that were parsed before artwork tokens existed
 # (they carry now-expired signed artwork URLs); bump when the parsed image
 # representation changes again
-PARSED_ITEM_CACHE_CHECKSUM = "artwork_tokens_v1"
+PARSED_ITEM_CACHE_CHECKSUM = "artist_fallback_v1"
```
[the remark this answers is not in the document]
Comment (MarvinSchenkel) [reply music_assistant/providers/apple_music/constants.py]: Fixed in 73a3782f6 - the comment now describes the checksum's general purpose.
